Source/DafnyCore/Compilers/Compiler-js.cs

```diff
@@ -1317,12 +1341,21 @@ protected class ClassWriter : IClassWriter {
       } else if (e.Value is bool) {
         wr.Write((bool)e.Value ? "true" : "false");
       } else if (e is CharLiteralExpr) {
-        var v = (string)e.Value;
-        wr.Write("'{0}'", v == "\\0" ? "\\u0000" : v);  // JavaScript doesn't have a \0
+        var escaped = TranslateEscapes((string)e.Value);
+        if (UnicodeChars) {
+          wr.Write($"new _dafny.CodePoint('{escaped}'.codePointAt(0))");
```

Comment:
It's pretty magical that this works in JS

Reply:
Yeah it's the thing that both JS and Python do where there's no character type, only strings of length one. It looks a little weird but I don't know of a better way to structure this.